### ISSUE TITLE
Launcher gets stuck on "Sign in"

### ISSUE THREAD

**Issue by AlexAzumi** [RiskiVR/BSLegacyLauncher#1]
I've been trying to use the launcher (that looks great btw) but every time that I want to download the game it keeps getting stuck on the "Sign in" screen.

I've tried different versions, such as 1.0.0, 1.1.0 and 1.4.0 and keeps doing the same.
If I close the window the launcher crashes and the back button does nothing when clicked on.

This is the log file that I found:

```
[13:08:29.891][DepotDownloader.Library][Info] - Triggered login again?
[13:08:29.948][DepotDownloader.Library][Info] - Connecting to Steam3...
[13:08:29.948][DepotDownloader.Library][Debug] - aaaaaaa
[13:08:30.693][DepotDownloader.Library][Info] -  Done!
[13:08:30.694][DepotDownloader.Library][Info] - Logging 'th3nix009' into Steam3...
[13:08:30.770][DepotDownloader.Library][Info] - Successfully connected to Steam3 service!
[13:08:31.088][DepotDownloader.Library][Info] - Disconnected from Steam
[13:08:31.088][DepotDownloader.Library][Info] - This account is protected by Steam Guard.
[13:08:31.088][DepotDownloader.Library][Debug] - Steam Account has Two Factor Auth enabled!
[13:08:38.741][DepotDownloader.Library][Info] - Triggered login again?
[13:08:38.742][DepotDownloader.Library][Info] - Connecting to Steam3...
[13:08:38.742][DepotDownloader.Library][Debug] - aaaaaaa
[13:08:39.290][DepotDownloader.Library][Info] -  Done!
[13:08:39.291][DepotDownloader.Library][Info] - Logging 'th3nix009' into Steam3...
[13:08:39.292][DepotDownloader.Library][Info] - Successfully connected to Steam3 service!
[13:08:39.589][DepotDownloader.Library][Info] -  Done!
[13:08:39.589][DepotDownloader.Library][Info] - Using Steam3 suggested CellID: 115
[13:08:39.590][DepotDownloader.Library][Info] - You selected version 1.4.0:3532596684905902618
[13:08:39.624][DepotDownloader.Library][Info] - Got 433 licenses for account!
```

Unfortunately I cannot use this amazing tool that you created. Hope this gets fixed.

**Comment by RiskiVR** [comment]
How long did you give it to load?

**Comment by AlexAzumi** [comment]
> How long did you give it to load?

About 20 minutes

**Comment by RiskiVR** [comment]
I'm not sure why it's getting stuck for you. Try restarting your PC. If the issue persists, I'm not sure what the problem is.

**Comment by ddakebono** [comment]
There could be a problem in DepotDownloader that isn't being caught, might have to look into it a bit.

**Comment by AlexAzumi** [comment]
I restarted my PC and the launcher keeps getting stuck. If I open the launcher after crashing the "select version" button does not work.

**Comment by ddakebono** [comment]
Delete the generated Beat Saber folder, it should be empty anyways

**Comment by AlexAzumi** [comment]
> Delete the generated Beat Saber folder, it should be empty anyways

I did. The select version button worked again but still getting stuck on Sign in.

**Comment by RiskiVR** [comment]
We still don't know the cause of this issue. The only current fix is to restart the Launcher. We'll continue to investigate this issue for fixes or catches to prevent softlocks.

**Comment by AlexAzumi** [comment]
Thanks for you hard work. Unfortunately I still can't use the Launcher, but I've been downloading the game using the batch scripts with no problems whatsoever.
After downloading the game, I move the Beat Saber folder into the Launcher's root folder and then I can play the game. It's not as convenient as using the Launcher itself to download the game, but that's the only way that works for me.

**Comment by dbqt** [comment]
I'm having the same issue as Alex and my logs looks pretty much the same too, it stops after getting the licenses for the account. Restarting the PC / Launcher doesn't help.

**Comment by RiskiVR** [comment]
Update: We found the issue, an exception was being thrown by DepotDownloader and didn't appear in logs. It was also not being catched, therefore, it left the Launcher stuck on the Sign In screen. We're working on patching it for the next release!

**Comment by UrkiMimi** [comment]
Whats funny is that I just got this issue and that did make my day lol

**Comment by AlexAzumi** [comment]
Did it get patched on `v1.3.0`?

**Comment by JTHicks26** [comment]
@AlexAzumi I was having this issue on `v1.2.2` and `v1.3.0` is working for me.

**Comment by AlexAzumi** [comment]
I have just tried `v1.3.0` but the bug is still there (at least for me). The log files look pretty much the same

**Comment by RiskiVR** [comment]
Bono has been very busy with their own projects and still hasn't gotten to DD fixes yet. I'm so terribly sorry for the wait.

**Comment by AlexAzumi** [comment]
Don't worry. I'll wait patiently for the patch. Thanks for your hard work!

**Comment by IRBerryz** [comment]
still having the same issue on 1.3.2

**Comment by RiskiVR** [comment]
We did not announce it was fixed yet

**Comment by HAlbert-sys** [comment]
I found that this was caused by windows defender blocking the app without asking for permission first, try allowing it to see what happens

**Comment by RiskiVR** [comment]
Fixed in v1.4.0
